Breaking change docs: OpenApiVersion defaults to OpenApi3_2 in .NET 11 (#37260)
dotnet/AspNetCore.Docs: aspnetcore/breaking-changes/11/openapi-version-default-3-2.md

ms.date: 06/16/2026
---
# OpenApiVersion defaults to OpenApi3_2

The default value of <xref:Microsoft.AspNetCore.OpenApi.OpenApiOptions.OpenApiVersion> has changed from `OpenApiSpecVersion.OpenApi3_1` to `OpenApiSpecVersion.OpenApi3_2`. Apps that depend on OpenAPI 3.1 output without explicitly setting the version now produce OpenAPI 3.2 documents.

## Version introduced

.NET 11 Preview 6

## Previous behavior

Previously, `OpenApiOptions.OpenApiVersion` defaulted to `OpenApiSpecVersion.OpenApi3_1`. Calling <xref:Microsoft.Extensions.DependencyInjection.OpenApiServiceCollectionExtensions.AddOpenApi%2A> without explicitly setting a version produced an OpenAPI 3.1 document.

## New behavior

Starting in ASP.NET Core 11 Preview 6, `OpenApiOptions.OpenApiVersion` defaults to `OpenApiSpecVersion.OpenApi3_2`. Calling `AddOpenApi` without explicitly setting a version now produces an OpenAPI 3.2 document.

## Type of breaking change

This change is a [behavioral change](/dotnet/core/compatibility/categories#behavioral-change).

## Reason for change

> ASP.NET Core's OpenAPI support aims to always use the latest released version of the specification by default so that apps automatically benefit from new capabilities.

Reviewer: ```suggestion
ASP.NET Core's OpenAPI support uses the latest released version of the OpenAPI specification as the default so that apps benefit from new capabilities without additional configuration.
```
Avoided using "aims to always" and "automatically benefit" which carried a bit of vague asperationional marketing flavor.

## Recommended action

> In most cases, no action is needed. If you use downstream tooling that doesn't yet support OpenAPI 3.2, configure the version explicitly:

Reviewer: ```suggestion
In most cases, no action is needed. To support downstream tooling that doesn't yet handle OpenAPI 3.2, such as code generators or older SDKs, configure the version explicitly:
```
It might help to indicate what kind of scenario.  Is this better?

Reviewer 2: > such as code generators or older SDKs

I'm not sure this is accurate.

@mikekistler @baywet could you advise here please?

Reviewer 3: I'd remove "or older SDKs" since SDKs are typically understood in the narrow sense of calling the API, not in the larger sense of "building software" in this context
